lib/iris/tests/__init__.py

```diff
@@ -620,16 +623,21 @@ def _unique_id(self):
 
         """
         # Obtain a consistent ID for the current test.
-
         # NB. unittest.TestCase.id() returns different values depending on
         # whether the test has been run explicitly, or via test discovery.
         # For example:
         #   python tests/test_plot.py => '__main__.TestContourf.test_tx'
         #   ird -t => 'iris.tests.test_plot.TestContourf.test_tx'
-        bits = self.id().split('.')[-3:]
+        bits = self.id().split('.')
         if bits[0] == '__main__':
-            file_name = os.path.basename(sys.modules['__main__'].__file__)
+            floc = sys.modules['__main__'].__file__
+            path, file_name = os.path.split(os.path.abspath(floc))
             bits[0] = os.path.splitext(file_name)[0]
+            folder, location = os.path.split(path)
+            bits = [location] + bits
+            while location != 'iris':
```

Comment:
@marqh :bomb: This won't work for documentation [example_tests](https://github.com/SciTools/iris/pull/2162/files#diff-53068f7ead231d14ec56529c4837ca8eR2), `iris` isn't the only root namespace module in the testing context.

You'll need to account for `example_tests` also being a terminal module.